src/client/compass.ts

```diff
@@ -111,9 +111,7 @@ export async function unlinkCompassComponents(cloudId: string, ecosystemAppId: s
 }
 
 export async function getComponentByExternalAlias(input: GetComponentByExternalAliasInput): Promise<ComponentPayload> {
-  const { errors, data } = await graphqlGateway.compass
-    .asApp()
-    .getComponentByExternalAlias({ ...input, externalSource: EXTERNAL_SOURCE });
+  const { errors, data } = await graphqlGateway.compass.asApp().getComponentByExternalAlias({ ...input });
```

Comment:
This is a refactor necessary to resolve a type error.

In this case, `getComponentByExternalAlias` is a custom function (defined on line 113) that calls the SDK’s `graphqlGateway.compass.asApp().getComponentByExternalAlias()` method internally, and has `input: GetComponentByExternalAliasInput` as parameter. In this input type, `externalSource` is non-nullable so whoever invokes this custom `getComponentByExternalAlias` function needs to provide the `externalSource`, hence we moved the line `externalSource: EXTERNAL_SOURCE` to `src/services/fetch-projects.ts:58`, where the function is invoked.